go-controller/pkg/node/gateway.go

```diff
@@ -143,16 +205,43 @@ func (g *gateway) Init(wf factory.NodeWatchFactory) error {
 	_, err = wf.AddServiceHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			svc := obj.(*kapi.Service)
-			g.AddService(svc)
+			err := wait.PollImmediate(500*time.Millisecond, 60*time.Second, func() (bool, error) {
```

Comment:
oh, I did not expect to see us adding a rettry loop here, but that seems okay.

Reply:
yes @trozet this looks to be holding event thread for 60s, let's use object retry [mechanism](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/obj_retry.go) for service and endpoint slice types too. sure, if you want we can wait until https://github.com/ovn-org/ovn-kubernetes/pull/2813 merged and then revisit this PR.